src/common/AndroidUtils.ts

```diff
@@ -309,21 +308,10 @@ export class AndroidSDKUtils {
     }
 
     public static hasEmulator(emulatorName: string): Promise<boolean> {
-        return new Promise((resolve, reject) => {
```

Comment:
There seems to be a discrepancy between our commands (and specially between iOS and Android platforms). Sometimes (particularly on iOS) we allow the user to provide their simulator by simply giving us its user friendly display name, and other times (particularly on Android) we don't work with display names and instead want IDs.

I made changes here and further down to make sure that a user can provide IDs or display names on both platforms, and we'll internally handle mapping a display name to an ID.